gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/VoltageCheckService.java

```diff
@@ -29,20 +32,35 @@
 @Service
 public class VoltageCheckService {
     private final Logger businessLogger;
+    private final UrlValidationService urlValidationService;
 
-    public VoltageCheckService(Logger businessLogger) {
+    public VoltageCheckService(Logger businessLogger, UrlValidationService urlValidationService) {
         this.businessLogger = businessLogger;
+        this.urlValidationService = urlValidationService;
     }
 
     public Optional<VoltageMonitoringResult> runVoltageCheck(SweData sweData, DichotomyResult<SweDichotomyValidationData> dichotomyResult, DichotomyDirection direction) {
 
         if ((direction == DichotomyDirection.ES_FR || direction == DichotomyDirection.FR_ES) && dichotomyResult.hasValidStep()) {
             businessLogger.info("Running voltage check");
-            Crac crac = sweData.getCracFrEs().getCrac();
-            Network network = NetworkUtil.getNetworkByDirection(sweData, direction);
-            VoltageMonitoring voltageMonitoring = new VoltageMonitoring(crac, network, dichotomyResult.getHighestValidStep().getRaoResult());
-            return Optional.of(voltageMonitoring.run(LoadFlow.find().getName(), LoadFlowParameters.load(), 4));
+            try {
+                Crac crac = sweData.getCracFrEs().getCrac();
+                Network network = getNetworkWithPra(dichotomyResult);
+                VoltageMonitoring voltageMonitoring = new VoltageMonitoring(crac, network, dichotomyResult.getHighestValidStep().getRaoResult());
+                return Optional.of(voltageMonitoring.run(LoadFlow.find().getName(), LoadFlowParameters.load(), 4));
+            } catch (Exception e) {
+                businessLogger.error("Exception during voltage check : {}", e.getMessage());
+            }
         }
         return Optional.empty();
```

Comment:
Can we use a "return early" pattern ? You invert the condition at the beginning to have this return optional right after. If the condition is not met, you carry on with the logic. This reduce the nesting level for the main logic.

Reply:
ok good idea